R/ncaa_bb_pbp.R

```diff
@@ -0,0 +1,677 @@
+          rvest::html_text(trim = T) |>
+          lubridate::mdy(tz = 'America/New_York', quiet = T),
+        T ~ game_start_datetime
+      ),
+      game_start_date = game_start_datetime |>
+        as.Date(tz = 'America/New_York'),
+      game_start_time = game_start_datetime |>
+        (\(x) paste0(
+          lubridate::hour(x),
+          ':',
+          sprintf("%02d", lubridate::minute(x)),
+          ' ET'
+        ))(),
+      game_start_time = dplyr::case_when(
+        is.na(
+          html |>
+            rvest::html_node(xpath = '/html/body/div[2]/table[3]/tr[1]/td[2]') |>
+            rvest::html_text(trim = T) |>
+            lubridate::mdy_hm(tz = 'America/New_York', quiet = T)
+        ) ~ NA_character_,
+        T ~ game_start_time
+      ),
+      away_team = html |>
+        rvest::html_nodes(xpath = '/html/body/div[2]/table[6]/tr[1]/td[2]') |>
+        rvest::html_text(),
+      home_team = html |>
+        rvest::html_nodes(xpath = '/html/body/div[2]/table[6]/tr[1]/td[4]') |>
+        rvest::html_text(),
+      away_team_id = html |>
+        rvest::html_node('body') |>
+        rvest::html_node('div#contentarea') |>
+        rvest::html_node('table.mytable') |>
+        rvest::html_nodes('tr') |>
+        rvest::html_nodes('a') |>
+        rvest::html_attr('href') |>
+        dplyr::nth(2) |>
+        (\(x) gsub('/teams/', '', x))() |>
+        as.character(),
+      home_team_id = html |>
+        rvest::html_node('body') |>
+        rvest::html_node('div#contentarea') |>
+        rvest::html_node('table.mytable') |>
+        rvest::html_nodes('tr') |>
+        rvest::html_nodes('a') |>
+        rvest::html_attr('href') |>
+        dplyr::nth(1) |>
+        (\(x) gsub('/teams/', '', x))() |>
+        as.character(),
+      #' check for which team has possession on each play -- this is effectively
+      #' the instantaneous "who has possession" whenever a given stat is recorded
+      #' e.g. if team A makes a shot, that shot is recorded as the possession of
+      #' team A, not team B who has possession immediately after the made shot
+      #' however, if team B forces a steal, team B is recorded as the possessing
+      #' team
+      pos_team = dplyr::case_when(
+        pbp_version == 'V2' ~ dplyr::case_when(
+          grepl(
+            ' jumpball won| assist| 2pt| 3pt| rebound| freethrow| foulon| steal',
+            X2
+          ) ~ away_team,
+          grepl(
+            ' jumpball won| assist| 2pt| 3pt| rebound| freethrow| foulon| steal',
+            X4
+          ) ~ home_team,
+          grepl(' jumpball lost| turnover', X2) ~ home_team,
+          grepl(' jumpball lost| turnover', X4) ~ away_team,
+          T ~ NA_character_
+        ),
+        T ~ dplyr::case_when(
+          grepl(' Rebound| made| missed| Assist| Steal',
+                X2) ~ away_team,
+          grepl(' Rebound| made| missed| Assist| Steal', X4) ~ home_team,
```

Comment:
Nitpick: may want to make these strings case-insensitive and add spaces after them to avoid picking up player names in the string